proto/substrait/algebra.proto

```diff
@@ -482,6 +482,13 @@ message Expression {
   message Cast {
     Type type = 1;
     Expression input = 2;
+    ConditionFlag condition_flag = 3;
+
+    enum ConditionFlag {
+      CONDITION_FLAG_UNSPECIFIED = 0;
+      CONDITION_FLAG_NULL_ON_FAILURE = 1;
+      CONDITION_FLAG_THROW_ON_FAILURE = 2;
```

Comment:
Curious why do we need both NULL_ON_FAILURE and THROW_ON_FAILURE. Since these are mutually exclusive, I expect that only one would be sufficient.

Reply:
When do conversion, if the `nullOnFailures= true` in velox, we can use `substraitCastExpr.set_condition_flag(::substrait::CONDITION_FLAG_NULL_ON_FAILURE)`.
if the `nullOnFailures = false`, we can use `substraitCastExpr.set_condition_flag(::substrait::CONDITION_FLAG_THROW_ON_FAILURE)`.
If we only have this one variation, we can use `CONDITION_FLAG_UNSPECIFIED` for the case `nullOnFailures = false`, but I am afraid this will be confused when there have other variations. Your thought?

Reply:
It's probably best to keep it explicit for clarity and stipulate the default of `THROW` in the prose in the spec.

Reply:
Change it like below.

``` 
   enum FailureBehavior {
      // throw an exception as default failure behavior
      FAILURE_BEHAVIOR_UNSPECIFIED = 0;
      FAILURE_BEHAVIOR_RETURN_NULL = 1;
    }
```
 Here we just use the default `FAILURE_BEHAVIOR_UNSPECIFIED = 0` represents the `THROW` cause the default(the first) in `enum` must end with `_UNSPECIFIED`. would this be ok?
Or we must use like below?
``` 
   enum FailureBehavior {
      FAILURE_BEHAVIOR_UNSPECIFIED = 0;
      FAILURE_BEHAVIOR_THROW_EXCEPTION = 1;
      FAILURE_BEHAVIOR_RETURN_NULL = 2;
    }
```

Reply:
Let's use

```protobuf
   enum FailureBehavior {
      FAILURE_BEHAVIOR_UNSPECIFIED = 0;
      FAILURE_BEHAVIOR_THROW_EXCEPTION = 1;
      FAILURE_BEHAVIOR_RETURN_NULL = 2;
   }
```

The `_UNSPECIFIED` suffixes are exactly that, they indicate that there is no error behavior specified. The default behavior should be one of the enum variants that is not `_UNSPECIFIED`.

Reply:
Got it. Thanks!
Changed this in the latest commit.